src/selectors/fridge.js

```diff
@@ -92,10 +91,18 @@ export const selectBreaches = createSelector(
   (fromDate, toDate, fridge) => {
     const { breaches } = fridge ?? {};
     const breachesInDateRange = breaches?.filtered(
-      'startTimestamp >= $0 && endTimestamp <= $0',
-      fromDate
+      '(startTimestamp <= $0 && (endTimestamp <= $1 || endTimestamp == null)) || ' +
+        '(startTimestamp >= $0 && endTimestamp <= $1)',
+      fromDate,
+      toDate
     );
 
-    return breachesInDateRange;
+    const adjustedBreaches = breachesInDateRange.map(({ timestamp, id, temperature }) => ({
+      id,
+      temperature,
+      timestamp: moment(timestamp).isBefore(moment(fromDate)) ? fromDate : timestamp,
+    }));
```

Comment:
this is adjusting the breaches where the start time is before the `fromDate`, so that in the chart there isn't an `x` value equal to the start of the breach, but all the timestamps start at `x+1000`, giving a really weird chart.
Not sure how clear this is - does it need a comment?

- fromDate = `x`
- earliest `breach.startTime = `y1`, `breach.endTime = `y2`
- second earliest `breach.startTime = z1`, `breach.endTime = z2`

Then `x >= y2` (otherwise the breach is not in the range), and `y2 > z1`, so `x >= z1`, and `y1` can safely be adjusted to `x`

Reply:
I think the code is clear... explanation makes sense, but the `x`, `y`, `z` variables are a bit confusing in the context of a time series :confused:.